src/java.base/linux/classes/jdk/internal/platform/cgroupv1/CgroupV1Subsystem.java

```diff
@@ -158,41 +158,37 @@ private static void createSubSystemController(CgroupV1Subsystem subsystem, Strin
      * setSubSystemPath based on the contents of /proc/self/cgroup
      */
     private static void setSubSystemControllerPath(CgroupV1Subsystem subsystem, String[] entry) {
-        String controllerName;
-        String base;
-        CgroupV1SubsystemController controller = null;
-        CgroupV1SubsystemController controller2 = null;
+        String controllerName = entry[1];
+        String base = entry[2];
 
-        controllerName = entry[1];
-        base = entry[2];
         if (controllerName != null && base != null) {
-            switch (controllerName) {
-                case "memory":
-                    controller = subsystem.memoryController();
-                    break;
-                case "cpuset":
-                    controller = subsystem.cpuSetController();
-                    break;
-                case "cpu,cpuacct":
-                case "cpuacct,cpu":
-                    controller = subsystem.cpuController();
-                    controller2 = subsystem.cpuAcctController();
-                    break;
-                case "cpuacct":
-                    controller = subsystem.cpuAcctController();
-                    break;
-                case "cpu":
-                    controller = subsystem.cpuController();
-                    break;
-                case "blkio":
-                    controller = subsystem.blkIOController();
-                    break;
-                // Ignore subsystems that we don't support
-                default:
-                    break;
+            for (String cName: controllerName.split(",")) {
```

Comment:
The previous logic checked for base!=null.  Do you think this is no longer necessary?

Reply:
Isn't this what line 164 does?

Reply:
Sorry, I missed the fact that line 164 was being retained.  I just saw a lot of line removal.  Looks good.

Reply:
No worries. Thank you for the review!